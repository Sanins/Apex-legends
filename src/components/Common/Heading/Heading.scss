.heading {
    &__h1 {
        font-size: 48px;
    }
    &__h2 {
        font-size: 42px;
    }
    &__h3 {
        font-size: 30px;
    }
}