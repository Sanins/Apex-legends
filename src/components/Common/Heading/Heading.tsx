import * as React from 'react';

import './Heading.scss';

interface HeadingProps {
    children: any;
    headingType: number;
}
export default function Heading(props: HeadingProps) {

	return(
        <>
            {props.headingType === 1 &&
                <h1 className='heading__h1'>
                    {props.children}
                </h1>
            }
            {props.headingType === 2 &&
                <h2 className='heading__h2'>
                    {props.children}
                </h2>
            }
            {props.headingType === 3 &&
                <h3 className='heading__h3'>
                    {props.children}
                </h3>
            }
            {props.headingType === 4 &&
                <h4 className='heading__h4'>
                    {props.children}
                </h4>
            }
            {props.headingType === 5 &&
                <h5 className='heading__h5'>
                    {props.children}
                </h5>
            }
            {props.headingType === 6 &&
                <h6 className='heading__h6'>
                    {props.children}
                </h6>
            }
        </>
	);
};

