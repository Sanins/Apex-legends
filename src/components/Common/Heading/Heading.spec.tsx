/* eslint-disable no-undef */
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Heading from "./Heading";

describe("<Heading>", () => {
	it("renders as h1", () => {
		const wrapper = shallow(
			<Heading headingType={1}>Heading</Heading>
		);

		expect(toJson(wrapper)).toMatchSnapshot();
		expect(wrapper.hasClass("heading__h1"));
		expect(wrapper.text()).toBe("Heading");
	});

	it("renders as h2", () => {
		const wrapper = shallow(
			<Heading headingType={2}>Heading</Heading>,
		);

		expect(toJson(wrapper)).toMatchSnapshot();
		expect(wrapper.hasClass("heading__h2"));
		expect(wrapper.text()).toBe("Heading");
	});

	it("renders as h3", () => {
		const wrapper = shallow(
			<Heading headingType={3}>Heading</Heading>,
		);

		expect(toJson(wrapper)).toMatchSnapshot();
		expect(wrapper.hasClass("heading__h2"));
		expect(wrapper.text()).toBe("Heading");
	});

	it("renders as h4", () => {
		const wrapper = shallow(
			<Heading headingType={4}>Heading</Heading>,
		);

		expect(toJson(wrapper)).toMatchSnapshot();
		expect(wrapper.hasClass("heading__h2"));
		expect(wrapper.text()).toBe("Heading");
	});

	it("renders as h5", () => {
		const wrapper = shallow(
			<Heading headingType={5}>Heading</Heading>,
		);

		expect(toJson(wrapper)).toMatchSnapshot();
		expect(wrapper.hasClass("heading__h2"));
		expect(wrapper.text()).toBe("Heading");
	});

	it("renders as h6", () => {
		const wrapper = shallow(
			<Heading headingType={6}>Heading</Heading>,
		);

		expect(toJson(wrapper)).toMatchSnapshot();
		expect(wrapper.hasClass("heading__h2"));
		expect(wrapper.text()).toBe("Heading");
	});
});
