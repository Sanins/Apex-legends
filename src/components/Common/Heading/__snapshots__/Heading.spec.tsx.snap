// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading> renders as h1 1`] = `
"<Fragment>
  <h1 className=\\"heading__h1\\">
    Heading
  </h1>
</Fragment>"
`;

exports[`<Heading> renders as h2 1`] = `
"<Fragment>
  <h2 className=\\"heading__h2\\">
    Heading
  </h2>
</Fragment>"
`;

exports[`<Heading> renders as h3 1`] = `
"<Fragment>
  <h3 className=\\"heading__h3\\">
    Heading
  </h3>
</Fragment>"
`;

exports[`<Heading> renders as h4 1`] = `
"<Fragment>
  <h4 className=\\"heading__h4\\">
    Heading
  </h4>
</Fragment>"
`;

exports[`<Heading> renders as h5 1`] = `
"<Fragment>
  <h5 className=\\"heading__h5\\">
    Heading
  </h5>
</Fragment>"
`;

exports[`<Heading> renders as h6 1`] = `
"<Fragment>
  <h6 className=\\"heading__h6\\">
    Heading
  </h6>
</Fragment>"
`;
